### PR TITLE
Add python oauth2 examples for cloud

### PR DESCRIPTION
**Pull request by zymap**

### CODE REVIEW

**Review by jiazhai — APPROVED** [review]

File: cloud/python/oauth2/produce-consume-oauth2-client.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- [OAuth2 Producer](./OAuth2Producer.py)
+- [OAuth2 Consumer](./OAuth2Consumer.py)
+
+## Steps
+
+1. Get the service URLs. For details, see [Get Pulsar service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).
+
+2. Get the oauth2 authentication parameters. For details, see [Get Oauth2 authentication parameters](https://github.com/streamnative/examples/tree/master/cloud#get-oauth2-authentication-parameters).
+
+3. Run a consume example to wait for the produced messages from the topic `oauth2`.
```
Comment (Huanli-Meng) [inline cloud/python/oauth2/produce-consume-oauth2-client.md:21]: 3. Run the Python consumer to receive messages from the topic `oauth2`.

File: cloud/python/oauth2/produce-consume-oauth2-client.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    ```shell
+    python3 OAuth2Consumer.py \
+        -su "pulsar+ssl://streamnative.cloud:6651 \
+        -t oauth2 -n 10 --auth-params '{
+        "issuer_url": "https://auth.streamnative.cloud",
+        "private_key": "/path/to/private.key",
+        "audience": "urn:sn:pulsar:test-organization-name:test-pulsar-instance-name"}'
+    ```
+
+4. Open another terminal, run a producer example to produce 10 messages to the pulsar topic `oauth2`.
```
Comment (Huanli-Meng) [inline cloud/python/oauth2/produce-consume-oauth2-client.md:32]:
```suggestion
4. Run the Python producer to publish messages to the topic `oauth2`.
```

File: cloud/python/oauth2/produce-consume-oauth2-client.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    python3 OAuth2Producer.py \
+        -su "pulsar+ssl://streamnative.cloud:6651 \
+        -t oauth2 -n 10 --auth-params '{
+        "issuer_url": "https://auth.streamnative.cloud",
+        "private_key": "/path/to/private.key",
+        "audience": "urn:sn:pulsar:test-organization-name:test-pulsar-instance-name"}'
+    }'
+    ```
+    
+    After running this producer example, you will see the following successful message.
```
Comment (Huanli-Meng) [inline cloud/python/oauth2/produce-consume-oauth2-client.md:44]:
```suggestion
    **Output**
```

**Review by Huanli-Meng — COMMENTED** [review]
leave my commmets. PTAL.

One more comments, i think the `produce-consume-oauth2-client.md` should be merged in the README doc.

File: cloud/python/oauth2/produce-consume-oauth2-client.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- [OAuth2 Producer](./OAuth2Producer.py)
+- [OAuth2 Consumer](./OAuth2Consumer.py)
+
+## Steps
+
+1. Get the service URLs. For details, see [Get Pulsar service URLs](https://github.com/streamnative/pulsar-examples/tree/master/cloud#get-pulsar-service-urls).
+
+2. Get the oauth2 authentication parameters. For details, see [Get Oauth2 authentication parameters](https://github.com/streamnative/examples/tree/master/cloud#get-oauth2-authentication-parameters).
+
+3. Run a consume example to wait for the produced messages from the topic `oauth2`.
```
Comment (zymap) [inline cloud/python/oauth2/produce-consume-oauth2-client.md:21]:
```suggestion
3. Run the Python consumer example to receive messages from the topic `oauth2`.
```

**Comment by zymap** [comment]
@Huanli-Meng I addressed your comments, PTAL. Thanks.

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Pull request merged**